ted2zim/multi/scraper.py

```diff
@@ -0,0 +1,330 @@
+                        playlist, mode="playlist"
+                    )
+                    log_run_result(success, process)
+            else:
+                logger.info("Falling back to ted2zim for playlist")
+                if not self.handle_single_zim(mode="playlist"):
+                    logger.error("ted2zim Failed")
+
+    def run_indiv_zim_mode(self, item, mode):
+        """ run ted2zim for an individual topic/playlist """
+
+        args = self.ted2zim_exe
+
+        if mode == "topic":
+            args += [
+                "--topics",
+                item,
+                "--output",
+                str(self.output_dir.joinpath("topics", item.replace(" ", "_"))),
+            ]
+        elif mode == "playlist":
+            args += [
+                "--playlist",
+                item,
+                "--output",
+                str(self.output_dir.joinpath("playlists", item)),
+            ]
+        else:
+            raise ValueError(f"Unsupported mode {mode}")
+
+        # set metadata args
+        metadata = self.metadata.get(item.replace(" ", "_"), {})
+        for key in (
+            "name",
+            "zim-file",
+            "title",
+            "description",
+            "tags",
+            "creator",
+        ):
+            # use value from metadata JSON if present else from command-line
+            value = metadata.get(
+                key, getattr(self, f"{key.replace('-', '_')}_format", None)
+            )
+
+            if value:  # only set arg if we have a value so it can be defaulted
+                args += [f"--{key}", self.compute_format(item, str(value))]
+
+        # ensure we supplied a name
+        if not has_argument("name", args):
+            args += ["--name", self.compute_format(item, self.name_format)]
+
+        # append regular ted2zim args
+        args += self.extra_args
+
+        if self.debug:
+            args += ["--debug"]
+
+        logger.debug(nicer_args_join(args))
+        process = subprocess.run(
+            args,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            universal_newlines=True,
+        )
+        return process.returncode == 0, process
+
+    def handle_single_zim(self, mode):
+        """ redirect request to standard ted2zim """
+
+        args = self.ted2zim_exe
+        if mode == "topic":
+            args += [
+                "--topics",
+                ",".join(self.topics),
+                "--output",
+                str(self.output_dir.joinpath("topics")),
+            ]
+        elif mode == "playlist":
+            args += [
+                "--playlist",
+                self.playlists[0],
+                "--output",
+                str(self.output_dir.joinpath("playlists")),
+            ]
+        else:
+            raise ValueError(f"Unsupported mode {mode}")
+        args += self.extra_args
+        if not ["--name" in item for item in self.extra_args] and self.name_format:
```

Comment:
All the `-format` should be reserved for `--indiz-zims`. this creates unnecessary complexity.
The fallback should be dumb: if not an indiv-zims request, forward to ted2zim.

Reply:
Am. Okay. I just tried to handle an edge case.